crowbar_framework/app/models/service_object.rb

```diff
@@ -1543,9 +1561,10 @@ def run_remote_chef_client(node, command, logfile_name)
 
   private
 
-  def wait_for_chef_clients(node)
-    unless RemoteNode.chef_ready?(node, 1200)
-      STDERR.puts "Waiting for already running chef-clients on #{node} failed"
+  def wait_for_chef_clients(node_name)
+    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.")
+    unless RemoteNode.chef_ready?(node_name, 1200)
```

Comment:
I probably wasn't clear enough: the code will work, the only issue is the logging. Because the code in `RemoteNode` is designed to be called from a fork, it's using `puts` for logging. Now, with the changes you're doing here, we would also use `RemoteNode` without a fork, and in that case, it's not appropriate to use `puts` for logging. It works, but it's not appropriate.